Add NoDelay property to TcpSocketClient

cmjacobs opened the pull request:
I saw there was an open issue requesting this (#101). I needed it too so I went ahead and gave it a shot. I added the Unit test and ran it on Windows Desktop. I have not had a chance to try it on other platforms.

cts-craig committed:
Added NoDelay property to TcpSocketClient

cts-craig committed:
Added Unit test for NoDelay on TcpSocketClient

cts-craig committed:
Added a missing comment

cts-craig committed:
Added NoDelay for TcpSocketListener

rdavisau commented:
Thanks for adding this! Haven't tested it yet but just browsing the changes it looks like the implementation of NoDelay on WinRT for `TcpSocketListener` isn't there. Is that because it's not possible on the platform? If so, can you add a note to the 'Platform considerations' section of `README.MD`? You can add yourself to the contributors list while you're there too. I'll try to test later this week and if we're all good make the release :+1:

cts-craig committed:
Added Test for TcpSocketListner.NoDelay
Added note to Platform Considerations in README.md regarding lack of support for NoDelay on TcpSocketListener under WinRT.

cts-craig committed:
Added myself to README.md

cmjacobs commented:
It appears that NoDelay was introduced on StreamSocketListener as of Win10. Please let me know if you need anything changed.

cmjacobs committed:
Fixed username in README.md (whoops)

cmjacobs commented:
It appears NoDelay wasn't added to StreamSocketListenerControl until Win10, oh well. Let me know if you need anything changed. (also, it appears I have been submitting changes as one username and commenting as another, sorry about that).
